### PR TITLE
Porting Arc<String> to Arc<str>

### PR DESCRIPTION
**Pull request by clarkmcc**

### CODE REVIEW

File: interpreter/src/functions.rs [hunk 1]
```diff
@@ -147,13 +147,16 @@ pub fn contains(This(this): This<Value>, arg: Value) -> Result<Value> {
...
+        Value::String(v) => Value::String(v),
         Value::Timestamp(t) => Value::String(t.to_rfc3339().into()),
         Value::Duration(v) => Value::String(format_duration(&v).into()),
         Value::Int(v) => Value::String(v.to_string().into()),
         Value::UInt(v) => Value::String(v.to_string().into()),
         Value::Float(v) => Value::String(v.to_string().into()),
-        Value::Bytes(v) => Value::String(Arc::new(String::from_utf8_lossy(v.as_slice()).into())),
+        Value::Bytes(v) => {
+            let str = String::from_utf8_lossy(v.as_slice()).to_string();
+            Value::String(Arc::from(str.as_str()))
```
Comment (junderw) [inline interpreter/src/functions.rs]:
```suggestion
            let str = String::from_utf8_lossy(v.as_slice());
            Value::String(Arc::from(str.as_ref()))
```

**Comment by junderw** [comment]
Also, in general, converting a string into an Arc<str> requires allocating the length of the string, but Arc<String> only requires allocating a fixed size 3 pointer wide struct inside the Arc.

So performance really depends on the average length of strings used in the wild... if your benchmarks use data representative of real world usage, then maybe Arc<String> is a better choice?

**Comment by clarkmcc** [comment]
@junderw it seems like the former would require cloning the string (re-allocating the length of the string), but doesn't the latter consume the string and require only a constant time allocation of the `Arc` itself?
```rust
Arc::from(String::from("Hello world").as_str());
Arc::from(String::from("Hello world"));
```

**Comment by junderw** [comment]
```rust
// 1.String::from performs one allocation of the length str.len()
// 2. and Arc::from performs another allocation that is the length of 2 AtomicUsizes + str.len()
Arc::from(String::from("Hello world").as_str());

// This only performs one allocation of the length of 2 AtomicUsizes + str.len()
Arc::from(String::from("Hello world"));
```

**Comment by junderw** [comment]
The only way to be 100% certain is to write some benchmarks that mimic real-world use.

**Pull request closed without merging**